The 'efi_uga' module is NOT supported on aarch64 (and other) architectures.
Attempting to load it causes a warning message to be displayed on boot, which
has to be manually dismissed by pressing ENTER.

Instead, perform a check on the CPU architecture, and only try to load this
module on i386 and x86_64.

diff '--color=auto' -apruN a/util/grub.d/00_header.in b/util/grub.d/00_header.in
--- a/util/grub.d/00_header.in	2025-11-18 11:49:31.000000000 +0000
+++ b/util/grub.d/00_header.in	2026-05-09 13:22:19.943910449 +0100
@@ -135,7 +135,9 @@ else # GRUB_FORCE_EFI_ALL_VIDEO is not s
     cat <<EOF
   if [ x\$grub_platform = xefi ]; then
     insmod efi_gop
-    insmod efi_uga
+    if [ x\$grub_cpu = xi386 -o x\$grub_cpu = xx86_64 ]; then
+      insmod efi_uga
+    fi
   elif [ x\$feature_all_video_module = xy ]; then
 EOF
 fi # end GRUB_FORCE_EFI_ALL_VIDEO
